lib/GoalCohort.js

```diff
@@ -269,12 +295,36 @@ GoalCohort.statics.joinCohort = async function(user, goal, code) {
     throw err;
   }
 
+  if (cohort.memberSlotsRemaining === 0) {
+    const err = new Error(`The team cannot accept any more members`);
+    err.status = 409;
+    throw err;
+  }
+
+  if (curCohort) {
+    const _user = curCohort.members.find(m => m.user.equals(user.id));
+    const _team = curCohort.teams[_user.teamIndex];
+
+    // if user is in a different team, leave the current team
+    if (_team.inviteCode !== code) {
```

Comment:
should these two blocks be combined? there should never be a reason to leave the same goal cohort twice subsequently here, right?

```js
if (
  _team.inviteCode !== code
 || (userCohort &&
      curCohort.nameCanonical !== canonicalize(userCohort.cohort))) {
  await goalCohortModel.leaveCohort(user, goal);
}
```